### ISSUE TITLE
How to handle confirmed action result to stay and display an error?

### ISSUE THREAD

**Issue by sneko** [jonatanklosko/material-ui-confirm#69]
Hi @jonatanklosko ,

Thanks for the library! Really often when the user confirms I do API requests and those can fail for any reason. I want to show the error to the user.

I could use a toaster/snackbar but it's not accessible.

So I would like to show an error block just above my text. I wanted to tweak the `confirmationButtonProps.onClick` to do my async request inside and depending on the result I would have tried modified the confirm content with an error block. Unfortunately my `onClick` is never called :(

Do you have an advice for this use case with your library?

Thank you,

---

EDIT: a solution otherwise would be to open a new modal to show the error. It makes things a bit weird, but when thinking about it "internal server error" should not happen often.

And also, let's say I have a list with options for each item (through a popover menu), in case **I don't use a confirm** to delete an item:
1. showing the error among the list would be weird and would break the display
2. showing the error above the list would require a scroll of maybe thousands of pixels, meaning the user would need to scroll down then to retry on the right item
3. and for sure, impossible to show the error into the popover menu

So maybe it just makes sense to use a new modal to harmonize with and without an action that has `useConfirm`.

---

EDIT2: despite this issue, I need to make the "confirm button" being disabled and having a loader while doing the async request, otherwise it's not obvious there is a request that can takes a long time.

**Comment by jonatanklosko** [comment]
Hey @sneko! This package has a rather specific scope, which is to streamline confirmation for typical operations. From what you described it looks like you'd ideally need more control over the dialog itself and the flow, so I would probably just go with a custom dialog. A while ago I played around with alternative approach, which you may find useful, more details in [this comment](https://github.com/jonatanklosko/material-ui-confirm/issues/22#issuecomment-743936227) :)

**Comment by sneko** [comment]
Hi @jonatanklosko ,

Thank you for pointing me to the other way. I reused your code and adjusted it so it complies with TypeScript checks (the issue was mainly about the context). I added an error alert is the `onConfirm` fails, and a loader on the confirm button.

If anyone wants to take a look to copy/paste, here it is:
https://github.com/inclusion-numerique/mediature/commit/5994266c1ae5e2b3b07d3523592432f38f3bd62d
